Jetty starts consuming CPU that remains high even without any traffic

**Jetty version(s)**
`9.4.41.v20210516`
`9.4.43.v20210629`

**Java version/vendor** `(use: java -version)`
```
$ java -version
openjdk version "1.8.0_292"
OpenJDK Runtime Environment (build 1.8.0_292-8u292-b10-0ubuntu1~18.04-b10)
OpenJDK 64-Bit Server VM (build 25.292-b10, mixed mode)
```
**OS type/version**
Ubuntu 18.04

**Description**
We had an issue with some of the production servers spiking with CPU use.
We do not know what triggered it. When affected servers are removed from load balancer and all traffic to them ceases, the CPU usage still remains high.

About 5 out of 8 servers got affected over a stretch of several hours. The servers belong to different clusters and run different application code on them. The common theme is that they all are Spring apps with embedded Jetty, are on Amazon EC2 and exposed to the internet via load balancers. They are in different AWS accounts, different load balancers and their load pattern is not correlated in any way. Just the AWS region is the same. We saw `9.4.41.v20210516` and `9.4.43.v20210629` affected. Restarting the app (with Jetty) fixes CPU usage.

We tried profiling the running app and building flamegraphs and saw that all the CPU is being eaten by a couple of "qtp" threads with stacktraces similar to:

```
"qtp708090483-69573" #69573 prio=5 os_prio=0 tid=0x00007f6b1e429000 nid=0x311f in Object.wait() [0x00007f6ac36c8000]
   java.lang.Thread.State: WAITING (on object monitor)
        at java.lang.Object.wait(Native Method)
        at java.lang.Object.wait(Object.java:502)
        at org.eclipse.jetty.server.HttpInput.blockForContent(HttpInput.java:584)
        at org.eclipse.jetty.server.HttpInput$1.blockForContent(HttpInput.java:1164)
        at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:330)
        - locked <0x00000000c54d9790> (a java.util.ArrayDeque)
        at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:270)
        at org.eclipse.jetty.util.UrlEncoded.decodeUtf8To(UrlEncoded.java:485)
        - locked <0x00000000c54dea40> (a org.eclipse.jetty.util.MultiMap)
        at org.eclipse.jetty.util.UrlEncoded.decodeTo(UrlEncoded.java:572)
        at org.eclipse.jetty.server.Request.extractFormParameters(Request.java:582)
        at org.eclipse.jetty.server.Request.extractContentParameters(Request.java:531)
        at org.eclipse.jetty.server.Request.getParameters(Request.java:435)
        at org.eclipse.jetty.server.Request.getParameterNames(Request.java:1093)
        at ch.qos.logback.access.spi.AccessEvent.buildRequestParameterMap(AccessEvent.java:338)
        at ch.qos.logback.access.spi.AccessEvent.getRequestParameterMap(AccessEvent.java:351)
        at ch.qos.logback.access.spi.AccessEvent.prepareForDeferredProcessing(AccessEvent.java:584)
        #
        # code from https://jira.qos.ch/browse/LOGBACK-1486 here
        #
        at ch.qos.logback.core.AsyncAppenderBase.append(AsyncAppenderBase.java:160)
        at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:84)
        at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:51)
        at ch.qos.logback.access.jetty.RequestLogImpl.log(RequestLogImpl.java:137)
        at org.eclipse.jetty.server.HttpChannel.onCompleted(HttpChannel.java:812)
        at org.eclipse.jetty.server.HttpChannel.onBadMessage(HttpChannel.java:872)
        at org.eclipse.jetty.server.HttpChannelOverHttp.badMessage(HttpChannelOverHttp.java:283)
        at org.eclipse.jetty.http.HttpParser.badMessage(HttpParser.java:1647)
        at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1629)
        at org.eclipse.jetty.server.HttpConnection.parseRequestBuffer(HttpConnection.java:376)
        at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:265)
        at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
        at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
        at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
        at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
        at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:383)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:882)
        at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1036)
        at java.lang.Thread.run(Thread.java:748)
```

Clearly we are at the end of request trying to log it (with `logback-access`) and while preparing request for deferred processing it tries to read some of the parameters from it and it all ends up in `HttpInput.read()`.

Which in turn blocks for content. However judging from how much CPU is used, our suspicion is that no actual sleeping happens, the thread just wakes up immediately, re-checks conditions, goes back to waiting and the story repeats.

I do not know anything about Jetty internals but given this code https://github.com/eclipse/jetty.project/blob/jetty-9.4.41.v20210516/jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java#L1161-L1166 just calls `input.blockForContent()` and always returns `true` (unless throws), then this loop https://github.com/eclipse/jetty.project/blob/jetty-9.4.41.v20210516/jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java#L314-L340 reduces to: 

```java
            while (true)
            {
                Content item = nextContent();
                if (item != null)
                {
                    l = get(item, b, off, len);
                    if (LOG.isDebugEnabled())
                        LOG.debug("{} read {} from {}", this, l, item);

                    // Consume any following poison pills
                    if (item.isEmpty())
                        nextInterceptedContent();
                    break;
                }

                // No content, so should we block?
                input.blockForContent();
            }
```

So in theory as long as `nextContent()` returns `null`, it just spins `input.blockForContent()`.

We tried verifying it by enabling debug logging with
```
<logger name="org.eclipse.jetty.server.HttpInput" level="DEBUG"/>
```
and it resulted in about 10Mb/sec of logs:
```
-----------------------------------------------------------------------------------------------------------------------------------------------
|       @timestamp        | level |       thread       |                                       message                                        |
|-------------------------|-------|--------------------|--------------------------------------------------------------------------------------|
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-71877 | HttpInputOverHTTP@1202b981[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
| 2021-10-08 14:54:35.000 | DEBUG | qtp708090483-69573 | HttpInputOverHTTP@72895282[c=0,q=0,[0]=null,s=STREAM] blocking for content timeout=0 |
-----------------------------------------------------------------------------------------------------------------------------------------------
```
So it does indeed look like `read()` just spins `blockForContent()` endlessly which for some reason wakes up every time but there are no data available (or `read()` would log something on this line https://github.com/eclipse/jetty.project/blob/jetty-9.4.41.v20210516/jetty-server/src/main/java/org/eclipse/jetty/server/HttpInput.java#L321 )
 
Remember, there is no any load on that server at that moment, no established HTTP connections, just couple of some old requests stuck in this state.

**How to reproduce?**
Unfortunately I have no idea what triggers it. We just saw several servers ended up in this state. Last time we saw it before today was several months ago. Given all affected servers are on AWS `eu-west-1` and behind load balancers, it could be that some region-wide issue with AWS load balancers resulted in some incorrect TCP connection closure or something like that which in turn triggered some edge condition in Jetty.


It also can be the same issue as https://github.com/eclipse/jetty.project/issues/5435 although I am not 100% sure

That reproduction is odd.

The looping I experience is actually caused by logback-access not being able to write the log file (the directory `/tmp/log/` doesn't exist on my system, and logback-access version 1.2.4 has a bug that keeps retrying infinitely).

When I upgrade logback-access to 1.2.6 I stop experiencing the the spin on attempting to create/write to the log file.
But then it just simply detects that it cannot create/write and essentially no-ops the Logger within logback-access.

If I change the `logback-access.xml` to have a sane directory location.

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<configuration debug="true">
    <appender name="file" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <file>logs/access.log</file>
        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
            <fileNamePattern>logs/access.%d{yyyy-MM-dd}.log</fileNamePattern>
        </rollingPolicy>

        <encoder>
            <pattern>combined</pattern>
        </encoder>
    </appender>

    <appender-ref ref="file"/>
</configuration>
```

And create the `logs/` directory within your example project then logback-access keeps running.

So this is now ...

* `<logback.version>1.2.6</logback.version>`
* `logback-access.xml` with sane directory locations
* `<jetty.version>9.4.41.v20210516<jetty.version>`

And yes, I can see the loop in the ManagedSelector.

Lets try upgrading Jetty real quick, as I would personally never run 9.4.41, as its' subject to security advisories - https://www.eclipse.org/jetty/security_reports.php

I upgrade to `<jetty.version>9.4.44.v20210927</jetty.version>` and try again.

I don't experience the looping in ManagedSelector.  Instead I notice that the HttpParser terminates.

```
06:05:02.900 [qtp611437735-48] DEBUG org.eclipse.jetty.server.HttpChannel - onCompleted for . written=50
06:05:02.907 [qtp611437735-48] WARN org.eclipse.jetty.server.Request - java.lang.IllegalStateException: Parser is terminated: HttpParser{s=CLOSE,0 of -1}
06:05:02,908 |-ERROR in ch.qos.logback.core.rolling.RollingFileAppender[file] - Appender [file] failed to append. org.eclipse.jetty.http.BadMessageException: 400: Unable to parse form content
	at org.eclipse.jetty.http.BadMessageException: 400: Unable to parse form content
	at 	at org.eclipse.jetty.server.Request.getParameters(Request.java:439)
	at 	at org.eclipse.jetty.server.Request.getParameterNames(Request.java:1093)
	at 	at ch.qos.logback.access.spi.AccessEvent.buildRequestParameterMap(AccessEvent.java:338)
	at 	at ch.qos.logback.access.spi.AccessEvent.getRequestParameterMap(AccessEvent.java:351)
	at 	at ch.qos.logback.access.spi.AccessEvent.prepareForDeferredProcessing(AccessEvent.java:584)
	at 	at ch.qos.logback.core.OutputStreamAppender.subAppend(OutputStreamAppender.java:223)
	at 	at ch.qos.logback.core.rolling.RollingFileAppender.subAppend(RollingFileAppender.java:235)
	at 	at ch.qos.logback.core.OutputStreamAppender.append(OutputStreamAppender.java:102)
	at 	at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:84)
	at 	at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:51)
	at 	at ch.qos.logback.access.jetty.RequestLogImpl.log(RequestLogImpl.java:137)
	at 	at org.eclipse.jetty.server.HttpChannel.onCompleted(HttpChannel.java:824)
	at 	at org.eclipse.jetty.server.HttpChannel.onBadMessage(HttpChannel.java:884)
	at 	at org.eclipse.jetty.server.HttpChannelOverHttp.badMessage(HttpChannelOverHttp.java:283)
	at 	at org.eclipse.jetty.http.HttpParser.badMessage(HttpParser.java:1653)
	at 	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1635)
	at 	at org.eclipse.jetty.server.HttpConnection.parseRequestBuffer(HttpConnection.java:381)
	at 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:265)
	at 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at 	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at 	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at 	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalStateException: Parser is terminated: HttpParser{s=CLOSE,0 of -1}
	at 	at org.eclipse.jetty.server.HttpConnection.fillAndParseForContent(HttpConnection.java:322)
	at 	at org.eclipse.jetty.server.HttpInputOverHTTP.produceContent(HttpInputOverHTTP.java:33)
	at 	at org.eclipse.jetty.server.HttpInput.nextContent(HttpInput.java:382)
	at 	at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:316)
	at 	at org.eclipse.jetty.server.HttpInput.read(HttpInput.java:270)
	at 	at org.eclipse.jetty.util.UrlEncoded.decodeUtf8To(UrlEncoded.java:485)
	at 	at org.eclipse.jetty.util.UrlEncoded.decodeTo(UrlEncoded.java:572)
	at 	at org.eclipse.jetty.server.Request.extractFormParameters(Request.java:582)
	at 	at org.eclipse.jetty.server.Request.extractContentParameters(Request.java:531)
	at 	at org.eclipse.jetty.server.Request.getParameters(Request.java:434)
	at 	... 23 common frames omitted
```

This is better, and is part of the work done in commit 64a7dda6687295f31ffa61a668032e78da519513 for Issue #6491

This state was reached because, the request has no `Host` header.

```
06:17:05.447 [qtp1543727556-48] WARN org.example.jetty6973.HttpListener - onRequestFailure(Request(POST .)@16a011d8)
org.eclipse.jetty.http.BadMessageException: 400: No Host
	at org.eclipse.jetty.http.HttpParser.parseFields(HttpParser.java:1203)
	at org.eclipse.jetty.http.HttpParser.parseNext(HttpParser.java:1528)
	at org.eclipse.jetty.server.HttpConnection.parseRequestBuffer(HttpConnection.java:381)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:265)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

This would be classified as an early failure, found during the HttpParser steps of the request line and headers.
The call to `HttpChannel.onBadMessage` is Jetty informing the raw HttpChannel / HttpConnection that the request is wholly bad.
A simple call to `server.getErrorHandler().badMessageError()` is performed to get the "go away" text (a single small buffer), to write back and close the connection.
Then the Request / Response is sent to the `server.getRequestLog().log(request,response)` to log what happened.
At which point `logback-access` attempts to read from the request and causes this entire chain of events.

Good to know that newer versions of Jetty don't experience the loop.

The changes I made to your example project can be found at https://github.com/joakime/jetty-6973/commit/95866bafecb11df7b1552a19125bfeb6902ffe0c

But I still want to create a PR that prevents the `.getParameterNames()` and `.getInputStream()` style calls against the Request from ever reading from the request body, as that's just wrong for a `RequestLog` to be doing, and changes what actually happened, as logback-access is logging success and bad information, sheerly because it changes the state of the request/response objects in it's own actions.

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java
@@ -872,12 +872,6 @@ public abstract class HttpChannel implements Runnable, HttpOutput.Interceptor
         if (LOG.isDebugEnabled())
             LOG.debug("onCompleted for {} written={}", getRequest().getRequestURI(), getBytesWritten());
 
-        if (_requestLog != null)
-        {
-            _request.onRequestLog();
-            _requestLog.log(_request, _response);
-        }
-
         long idleTO = _configuration.getIdleTimeout();
         if (idleTO >= 0 && getIdleTimeout() != _oldIdleTimeout)
             setIdleTimeout(_oldIdleTimeout);

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java
@@ -873,7 +873,10 @@ public abstract class HttpChannel implements Runnable, HttpOutput.Interceptor
             LOG.debug("onCompleted for {} written={}", getRequest().getRequestURI(), getBytesWritten());
 
         if (_requestLog != null)
+        {
+            _request.onRequestLog();
             _requestLog.log(_request, _response);
+        }
 
         long idleTO = _configuration.getIdleTimeout();
         if (idleTO >= 0 && getIdleTimeout() != _oldIdleTimeout)

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
@@ -1413,20 +1413,6 @@ public class Request implements HttpServletRequest
         return session.getId();
     }
 
-    protected void onRequestLog()
-    {
-        // Don't allow pulling more parameters
-        _contentParamsExtracted = true;
-
-        // Reset the status code to what was committed
-        MetaData.Response committedResponse = getResponse().getCommittedMetaData();
-        if (committedResponse != null)
-        {
-            getResponse().setStatus(committedResponse.getStatus());
-            // TODO: Reset the response headers to what they were when committed
-        }
-    }
-
     /**
      * Called when the request is fully finished being handled.
      * For every session in any context that the session has
@@ -1434,6 +1420,23 @@ public class Request implements HttpServletRequest
      */
     public void onCompleted()
     {
+        RequestLog requestLog = getHttpChannel().getRequestLog();
+        if (requestLog != null)
+        {
+            // Don't allow pulling more parameters
+            _contentParamsExtracted = true;
+
+            // Reset the status code to what was committed
+            MetaData.Response committedResponse = getResponse().getCommittedMetaData();
+            if (committedResponse != null)
+            {
+                getResponse().setStatus(committedResponse.getStatus());
+                // TODO: Reset the response headers to what they were when committed
+            }
+
+            requestLog.log(this, getResponse());
+        }
+
         if (_sessions != null)
         {
             for (Session s:_sessions)

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
@@ -1413,6 +1413,20 @@ public class Request implements HttpServletRequest
         return session.getId();
     }
 
+    protected void onRequestLog()
+    {
+        // Don't allow pulling more parameters
+        _contentParamsExtracted = true;
+
+        // Reset the status code to what was committed
+        MetaData.Response committedResponse = getResponse().getCommittedMetaData();
+        if (committedResponse != null)
+        {
+            getResponse().setStatus(committedResponse.getStatus());
+            // TODO: Reset the headers code to what was committed
+        }
+    }
+
     /**
      * Called when the request is fully finished being handled.
      * For every session in any context that the session has

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
@@ -1423,7 +1423,7 @@ public class Request implements HttpServletRequest
         if (committedResponse != null)
         {
             getResponse().setStatus(committedResponse.getStatus());
-            // TODO: Reset the headers code to what was committed
+            // TODO: Reset the response headers to what they were when committed
         }
     }
 

--- a/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
+++ b/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java
@@ -1420,21 +1420,27 @@ public class Request implements HttpServletRequest
      */
     public void onCompleted()
     {
-        RequestLog requestLog = getHttpChannel().getRequestLog();
-        if (requestLog != null)
+        HttpChannel httpChannel = getHttpChannel();
+        // httpChannel can be null in some scenarios
+        // it's not possible to use requestlog in those scenarios anyway.
+        if (httpChannel != null)
         {
-            // Don't allow pulling more parameters
-            _contentParamsExtracted = true;
-
-            // Reset the status code to what was committed
-            MetaData.Response committedResponse = getResponse().getCommittedMetaData();
-            if (committedResponse != null)
+            RequestLog requestLog = httpChannel.getRequestLog();
+            if (requestLog != null)
             {
-                getResponse().setStatus(committedResponse.getStatus());
-                // TODO: Reset the response headers to what they were when committed
-            }
+                // Don't allow pulling more parameters
+                _contentParamsExtracted = true;
 
-            requestLog.log(this, getResponse());
+                // Reset the status code to what was committed
+                MetaData.Response committedResponse = getResponse().getCommittedMetaData();
+                if (committedResponse != null)
+                {
+                    getResponse().setStatus(committedResponse.getStatus());
+                    // TODO: Reset the response headers to what they were when committed
+                }
+
+                requestLog.log(this, getResponse());
+            }
         }
 
         if (_sessions != null)

--- a/jetty-server/src/test/java/org/eclipse/jetty/server/RequestLogTest.java
+++ b/jetty-server/src/test/java/org/eclipse/jetty/server/RequestLogTest.java
@@ -142,7 +142,8 @@ public class RequestLogTest
     }
 
     /**
-     * Test an unread HTTP/1.1 POST (where the server doesn't read the POST, but the RequestLog attempts to read it instead).
+     * Test an unread HTTP/1.1 POST, it has valid body content, the dispatched Handler on the server doesn't read the POST body content.
+     * The RequestLog accidentally attempts to read the Request body content due to the use of Request.getParameterNames() API.
      */
     @Test
     public void testNormalPostFormRequest() throws Exception
@@ -223,7 +224,9 @@ public class RequestLogTest
 
     /**
      * Test a Bad HTTP/1.1 POST Request, it has body content, but also includes a Content-Length + Transfer-Encoding header.
-     * The server doesn't read the POST body, but the RequestLog attempts to read the request body.
+     * This results in a BadMessage internally, and the Handler is never called.
+     * The POST body content is never read by a Handler or the error handling code.
+     * The RequestLog accidentally attempts to read the Request body content due to the use of Request.getParameterNames() API.
      */
     @Test
     public void testBadPostFormRequest() throws Exception
@@ -306,7 +309,7 @@ public class RequestLogTest
     }
 
     /**
-     * Test where the response is committed, then changes the status code and response headers.
+     * Test where the response is committed, then the dispatch changes the status code and response headers.
      * The RequestLog should see the committed status code and committed headers, not the changed ones.
      */
     @Disabled("Support for restoring Committed Response Headers coming in later PR")

--- a/jetty-server/src/test/java/org/eclipse/jetty/server/RequestLogTest.java
+++ b/jetty-server/src/test/java/org/eclipse/jetty/server/RequestLogTest.java
@@ -1,0 +1,395 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.jetty.server.handler.AbstractHandler;
+import org.eclipse.jetty.util.BlockingArrayQueue;
+import org.eclipse.jetty.util.IO;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Test raw behaviors of RequestLog and how Request / Response objects behave during
+ * the call to a RequestLog implementation.
+ * <p>
+ * This differs from other RequestLog tests as it test the combination of
+ * bad requests, and RequestLog implementations that can cause changes
+ * in the request or response objects.
+ * </p>
+ */
+public class RequestLogTest
+{
+    private static final Logger LOG = LoggerFactory.getLogger(RequestLogTest.class);
+
+    private static class NormalResponse extends AbstractHandler
+    {
+        public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
+        {
+            response.setCharacterEncoding("UTF-8");
+            response.setContentType("text/plain");
+            response.getWriter().printf("Got %s to %s%n", request.getMethod(), request.getRequestURI());
+            baseRequest.setHandled(true);
+        }
+    }
+
+    public Server createServer(RequestLog requestLog, Handler serverHandler) throws Exception
+    {
+        Server server = new Server();
+        ServerConnector connector = new ServerConnector(server);
+        connector.setPort(0);
+        server.addConnector(connector);
+
+        server.setRequestLog(requestLog);
+        server.setHandler(serverHandler);
+        return server;
+    }
+
+    /**
+     * Test a normal GET request.
+     */
+    @Test
+    public void testNormalGetRequest() throws Exception
+    {
+        Server server = null;
+        try
+        {
+            BlockingArrayQueue<String> requestLogLines = new BlockingArrayQueue<>();
+
+            server = createServer((request, response1) ->
+                    requestLogLines.add(String.format("method:%s|uri:%s|status:%d", request.getMethod(), request.getRequestURI(), response1.getStatus())),
+                new NormalResponse());
+            server.start();
+
+            URI baseURI = server.getURI();
+
+            try (Socket socket = new Socket(baseURI.getHost(), baseURI.getPort());
+                 OutputStream out = socket.getOutputStream();
+                 InputStream in = socket.getInputStream())
+            {
+                StringBuilder req = new StringBuilder();
+                req.append("GET /hello HTTP/1.1\r\n");
+                req.append("Host: ").append(baseURI.getRawAuthority()).append("\r\n");
+                req.append("Connection: close\r\n");
+                req.append("\r\n");
+
+                byte[] bufRequest = req.toString().getBytes(UTF_8);
+
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Request--\n" + req);
+                out.write(bufRequest);
+                out.flush();
+
+                ByteArrayOutputStream outBuf = new ByteArrayOutputStream();
+                IO.copy(in, outBuf);
+                String response = outBuf.toString(UTF_8);
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Response--\n" + response);
+
+                List<String> responseLines = response.lines()
+                    .map(String::trim)
+                    .collect(Collectors.toList());
+
+                // Find status code
+                String responseStatusLine = responseLines.get(0);
+                assertThat("Status Code Response", responseStatusLine, containsString("HTTP/1.1 200"));
+
+                // Find body content (always last line)
+                String bodyContent = responseLines.get(responseLines.size() - 1);
+                assertThat("Body Content", bodyContent, containsString("Got GET to /hello"));
+
+                String reqlog = requestLogLines.poll(5, TimeUnit.SECONDS);
+                assertThat("RequestLog", reqlog, containsString("method:GET|uri:/hello|status:200"));
+            }
+        }
+        finally
+        {
+            LifeCycle.stop(server);
+        }
+    }
+
+    /**
+     * Test an unread HTTP/1.1 POST (where the server doesn't read the POST, but the RequestLog attempts to read it instead).
+     */
+    @Test
+    public void testNormalPostFormRequest() throws Exception
+    {
+        Server server = null;
+        try
+        {
+            BlockingArrayQueue<String> requestLogLines = new BlockingArrayQueue<>();
+
+            // Use a Servlet API that would cause a read of the Request inputStream.
+            // This should result in no paramNames, as nothing is read during RequestLog execution
+            server = createServer((request, response1) ->
+            {
+                // Use a Servlet API that would cause a read of the Request inputStream.
+                List<String> paramNames = Collections.list(request.getParameterNames());
+                // This should result in no paramNames, as nothing is read during RequestLog execution
+                requestLogLines.add(String.format("method:%s|uri:%s|paramNames.size:%d|status:%d", request.getMethod(), request.getRequestURI(), paramNames.size(), response1.getStatus()));
+            }, new NormalResponse());
+            server.start();
+
+            URI baseURI = server.getURI();
+
+            try (Socket socket = new Socket(baseURI.getHost(), baseURI.getPort());
+                 OutputStream out = socket.getOutputStream();
+                 InputStream in = socket.getInputStream())
+            {
+                StringBuilder form = new StringBuilder();
+                for (int i = 'a'; i < 'j'; i++)
+                {
+                    form.append((char)i).append("=").append(i).append("&");
+                }
+
+                byte[] bufForm = form.toString().getBytes(UTF_8);
+
+                StringBuilder req = new StringBuilder();
+                req.append("POST /hello HTTP/1.1\r\n");
+                req.append("Host: ").append(baseURI.getRawAuthority()).append("\r\n");
+                req.append("Content-Type: ").append(MimeTypes.Type.FORM_ENCODED).append("\r\n");
+                req.append("Content-Length: ").append(bufForm.length).append("\r\n");
+                req.append("Connection: close\r\n");
+                req.append("\r\n");
+
+                byte[] bufRequest = req.toString().getBytes(UTF_8);
+
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Request--\n" + req);
+                out.write(bufRequest);
+                out.write(bufForm);
+                out.flush();
+
+                ByteArrayOutputStream outBuf = new ByteArrayOutputStream();
+                IO.copy(in, outBuf);
+                String response = outBuf.toString(UTF_8);
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Response--\n" + response);
+
+                List<String> responseLines = response.lines()
+                    .map(String::trim)
+                    .collect(Collectors.toList());
+
+                // Find status code
+                String responseStatusLine = responseLines.get(0);
+                assertThat("Status Code Response", responseStatusLine, containsString("HTTP/1.1 200"));
+
+                // Find body content (always last line)
+                String bodyContent = responseLines.get(responseLines.size() - 1);
+                assertThat("Body Content", bodyContent, containsString("Got POST to /hello"));
+
+                String reqlog = requestLogLines.poll(5, TimeUnit.SECONDS);
+                assertThat("RequestLog", reqlog, containsString("method:POST|uri:/hello|paramNames.size:0|status:200"));
+            }
+        }
+        finally
+        {
+            LifeCycle.stop(server);
+        }
+    }
+
+    /**
+     * Test a Bad HTTP/1.1 POST Request, it has body content, but also includes a Content-Length + Transfer-Encoding header.
+     * The server doesn't read the POST body, but the RequestLog attempts to read the request body.
+     */
+    @Test
+    public void testBadPostFormRequest() throws Exception
+    {
+        Server server = null;
+        try
+        {
+            BlockingArrayQueue<String> requestLogLines = new BlockingArrayQueue<>();
+
+            // Use a Servlet API that would cause a read of the Request inputStream.
+            // This should result in no paramNames, as nothing is read during RequestLog execution
+            server = createServer((request, response1) ->
+            {
+                // Use a Servlet API that would cause a read of the Request inputStream.
+                List<String> paramNames = Collections.list(request.getParameterNames());
+                // This should result in no paramNames, as nothing is read during RequestLog execution
+                requestLogLines.add(String.format("method:%s|uri:%s|paramNames.size:%d|status:%d", request.getMethod(), request.getRequestURI(), paramNames.size(), response1.getStatus()));
+            }, new NormalResponse());
+            server.start();
+
+            URI baseURI = server.getURI();
+
+            try (Socket socket = new Socket(baseURI.getHost(), baseURI.getPort());
+                 OutputStream out = socket.getOutputStream();
+                 InputStream in = socket.getInputStream())
+            {
+                StringBuilder form = new StringBuilder();
+                for (int i = 'a'; i < 'j'; i++)
+                {
+                    form.append((char)i).append("=").append(i).append("&");
+                }
+
+                byte[] bufForm = form.toString().getBytes(UTF_8);
+
+                StringBuilder req = new StringBuilder();
+                req.append("POST /hello HTTP/1.1\r\n");
+                req.append("Host: ").append(baseURI.getRawAuthority()).append("\r\n");
+                req.append("Content-Type: ").append(MimeTypes.Type.FORM_ENCODED).append("\r\n");
+                req.append("Content-Length: ").append(bufForm.length).append("\r\n");
+                // add extra Transfer-Encoding: chunked header, making the POST request invalid per HTTP spec
+                req.append("Transfer-Encoding: chunked\r\n");
+                req.append("Connection: close\r\n");
+                req.append("\r\n");
+
+                byte[] bufRequest = req.toString().getBytes(UTF_8);
+
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Request--\n" + req);
+                out.write(bufRequest);
+                out.write(bufForm);
+                out.flush();
+
+                ByteArrayOutputStream outBuf = new ByteArrayOutputStream();
+                IO.copy(in, outBuf);
+                String response = outBuf.toString(UTF_8);
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Response--\n" + response);
+
+                List<String> responseLines = response.lines()
+                    .map(String::trim)
+                    .collect(Collectors.toList());
+
+                // Find status code
+                String responseStatusLine = responseLines.get(0);
+                assertThat("Status Code Response", responseStatusLine, containsString("HTTP/1.1 400 Bad Request"));
+
+                // Find body content (always last line)
+                String bodyContent = responseLines.get(responseLines.size() - 1);
+                assertThat("Body Content", bodyContent, containsString("reason: Transfer-Encoding and Content-Length"));
+
+                // We should see a requestlog entry for this 400 response
+                String reqlog = requestLogLines.poll(3, TimeUnit.SECONDS);
+                assertThat("RequestLog", reqlog, containsString("method:POST|uri:/hello|paramNames.size:0|status:400"));
+            }
+        }
+        finally
+        {
+            LifeCycle.stop(server);
+        }
+    }
+
+    /**
+     * Test where the response is committed, then changes the status code and response headers.
+     * The RequestLog should see the committed status code and committed headers, not the changed ones.
+     */
+    @Disabled("Support for restoring Committed Response Headers coming in later PR")
+    @Test
+    public void testResponseThenChangeStatusAndHeaders() throws Exception
+    {
+        Server server = null;
+        try
+        {
+            BlockingArrayQueue<String> requestLogLines = new BlockingArrayQueue<>();
+
+            RequestLog requestLog = (request, response) ->
+            {
+                String xname = response.getHeader("X-Name");
+                requestLogLines.add(String.format("method:%s|uri:%s|header[x-name]:%s|status:%d", request.getMethod(), request.getRequestURI(), xname, response.getStatus()));
+            };
+
+            Handler handler = new AbstractHandler()
+            {
+                @Override
+                public void handle(String target, Request baseRequest, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException
+                {
+                    response.setStatus(202);
+                    response.setHeader("X-Name", "actual");
+                    response.setCharacterEncoding("UTF-8");
+                    response.setContentType("text/plain");
+                    response.getWriter().printf("Got %s to %s%n", request.getMethod(), request.getRequestURI());
+                    response.flushBuffer();
+                    assertTrue(response.isCommitted(), "Response should be committed");
+                    baseRequest.setHandled(true);
+                    response.setStatus(204);
+                    response.setHeader("X-Name", "post-commit");
+                }
+            };
+
+            server = createServer(requestLog, handler);
+            server.start();
+
+            URI baseURI = server.getURI();
+
+            try (Socket socket = new Socket(baseURI.getHost(), baseURI.getPort());
+                 OutputStream out = socket.getOutputStream();
+                 InputStream in = socket.getInputStream())
+            {
+                StringBuilder req = new StringBuilder();
+                req.append("GET /world HTTP/1.1\r\n");
+                req.append("Host: ").append(baseURI.getRawAuthority()).append("\r\n");
+                req.append("Connection: close\r\n");
+                req.append("\r\n");
+
+                byte[] bufRequest = req.toString().getBytes(UTF_8);
+
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Request--\n" + req);
+                out.write(bufRequest);
+                out.flush();
+
+                ByteArrayOutputStream outBuf = new ByteArrayOutputStream();
+                IO.copy(in, outBuf);
+                String response = outBuf.toString(UTF_8);
+                if (LOG.isDebugEnabled())
+                    LOG.debug("--Response--\n" + response);
+
+                List<String> responseLines = response.lines()
+                    .map(String::trim)
+                    .collect(Collectors.toList());
+
+                // Find status code
+                String responseStatusLine = responseLines.get(0);
+                assertThat("Status Code Response", responseStatusLine, containsString("HTTP/1.1 202 Accepted"));
+
+                // Find body content (always last line)
+                String bodyContent = responseLines.get(responseLines.size() - 1);
+                assertThat("Body Content", bodyContent, containsString("Got GET to /world"));
+
+                // We should see a requestlog entry for this 400 response
+                String reqlog = requestLogLines.poll(3, TimeUnit.SECONDS);
+                assertThat("RequestLog", reqlog, containsString("method:GET|uri:/world|header[x-name]:actual|status:202"));
+            }
+        }
+        finally
+        {
+            LifeCycle.stop(server);
+        }
+    }
+}

--- a/jetty-server/src/test/resources/jetty-logging.properties
+++ b/jetty-server/src/test/resources/jetty-logging.properties
@@ -4,3 +4,4 @@
 #org.eclipse.jetty.servlet.LEVEL=DEBUG
 #org.eclipse.jetty.server.ConnectionLimit.LEVEL=DEBUG
 #org.eclipse.jetty.server.AcceptRateLimit.LEVEL=DEBUG
+#org.eclipse.jetty.server.RequestLogTest.LEVEL=DEBUG

